### ISSUE TITLE
[Bug / Feature Request] Cannot close files once opened

### ISSUE THREAD

**Issue by Checkroth** [pikepdf/pikepdf#37]
# Summary

When you call `pikepdf.open`, it opens an OS handler for the file. You can see this open with `lsof`

```
In [1]: import pikepdf

In [2]: p = pikepdf.open('pdf.pdf')
```

will create the handler:

```
python  17331 checkroth   12r      REG                8,1   433994 4750331 /home/checkroth/pdf.pdf
```

# The Problem

There is no way to _close_ a file with pikepdf after opening it. This is likely an edge case, but a high-usage system that is not frequently restarted will pile up dead file handlers until you eventually get to an `OS Error: Too many open files`.

# A temporary fix

You can manually delete the reference to a pikepdf.Pdf instance. From the above code,

```
In [3]: del p
```

will delete the file handler as well.

# The Feature Request

It would be great to have a `Pdf.close()` function that will handle this safely in the context of PikePDF.

**Comment by jbarlow83** [comment]
Similar to #31

**Comment by Checkroth** [comment]
Specifically this comment: https://github.com/pikepdf/pikepdf/issues/31#issuecomment-474189778

In my particular use case, I'm saving PDFs through Pike as a temp file and then moving them to replace other files with shutil. Checking lsof after some time reveals that there are a lot of (removed) temp file handlers left over until the service is eventually restarted.

It looks to be the same core issue as the editing permission issue outlined in that comment.

**Comment by jbarlow83** [comment]
Fixed for v1.2.0
